headerCheck/factory/coreComponents.go

```diff
@@ -0,0 +1,130 @@
+	ratingsConfig *data.RatingsConfig,
+	networkConfig *data.NetworkConfig,
+) (*coreComponents, error) {
+	marshalizer, err := marshalizerFactory.NewMarshalizer(marshalizerType)
+	if err != nil {
+		return nil, err
+	}
+
+	hasher, err := hasherFactory.NewHasher(hasherType)
+	if err != nil {
+		return nil, err
+	}
+
+	rater, err := createRater(ratingsConfig, networkConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &coreComponents{
+		Marshaller: marshalizer,
+		Hasher:     hasher,
+		Rater:      rater,
+	}, nil
+}
+
+func createRater(rc *data.RatingsConfig, nc *data.NetworkConfig) (nodesCoordinator.ChanceComputer, error) {
+	ratingsConfig := createRatingsConfig(rc)
+
+	ratingDataArgs := rating.RatingsDataArg{
+		Config:                   ratingsConfig,
+		ShardConsensusSize:       uint32(nc.ShardConsensusGroupSize),
+		MetaConsensusSize:        uint32(nc.MetaConsensusGroup),
+		ShardMinNodes:            uint32(nc.NumNodesInShard),
+		MetaMinNodes:             uint32(nc.NumMetachainNodes),
+		RoundDurationMiliseconds: uint64(nc.RoundDuration),
+	}
+
+	ratingsData, err := rating.NewRatingsData(ratingDataArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	rater, err := rating.NewBlockSigningRater(ratingsData)
+	if err != nil {
+		return nil, err
+	}
+
+	return rater, nil
+}
+
+func createRatingsConfig(rd *data.RatingsConfig) config.RatingsConfig {
+	selectionChances := make([]*config.SelectionChance, len(rd.GeneralSelectionChances))
+	for i, v := range rd.GeneralSelectionChances {
+		selectionChance := &config.SelectionChance{
+			MaxThreshold:  v.MaxThreshold,
+			ChancePercent: v.ChancePercent,
+		}
+		selectionChances[i] = selectionChance
+	}
+
+	general := config.General{
+		StartRating:           rd.GeneralStartRating,
+		MaxRating:             rd.GeneralMaxRating,
+		MinRating:             rd.GeneralMinRating,
+		SignedBlocksThreshold: rd.GetSignedBlockThreshold(),
+		SelectionChances:      selectionChances,
+	}
+
+	shardChain := config.ShardChain{
+		RatingSteps: config.RatingSteps{
+			HoursToMaxRatingFromStartRating: rd.ShardchainHoursToMaxRatingFromStartRating,
+			ProposerValidatorImportance:     rd.GetShardchainProposerValidatorImportance(),
+			ProposerDecreaseFactor:          rd.GetShardchainProposerDecreaseFactor(),
+			ValidatorDecreaseFactor:         rd.GetShardchainValidatorDecreaseFactor(),
+			ConsecutiveMissedBlocksPenalty:  rd.GetShardchainConsecutiveMissedBlocksPenalty(),
+		},
+	}
+
+	metaChain := config.MetaChain{
+		RatingSteps: config.RatingSteps{
+			HoursToMaxRatingFromStartRating: rd.MetachainHoursToMaxRatingFromStartRating,
+			ProposerValidatorImportance:     rd.GetMetachainProposerValidatorImportance(),
+			ProposerDecreaseFactor:          rd.GetMetachainProposerDecreaseFactor(),
+			ValidatorDecreaseFactor:         rd.GetMetachainValidatorDecreaseFactor(),
+			ConsecutiveMissedBlocksPenalty:  rd.GetMetachainConsecutiveMissedBlocksPenalty(),
+		},
+	}
+
+	peerHonesty := config.PeerHonestyConfig{
```

Comment:
I do not think this is needed. the peer honesty is used mainly in consensus and does not alter blockchain state

Reply:
ok. removed it :+1: